application/cms/page_service.py

```diff
@@ -145,6 +149,35 @@ def update_page(self, page, data, last_updated_by):
             self.logger.info(message)
             return page
 
+    def _set_data_sources(self, page, data_source_forms):
+        current_data_sources = page.data_sources
+        page.data_sources = []
+
+        for i, data_source_form in enumerate(data_source_forms):
```

Comment:
Are we guaranteed the ordering of `data_source_forms`?  Because if not then the counting by index isn't necessarily going to work for figuring out if it's an existing source, etc.

Reply:
OK I checked at `get_data_source_forms` and this looks fine.